### PR TITLE
enhance: Add open telemetry tracing for compaction

### LINKED ISSUE #30167: [Enhancement]: Add tracing for compaction
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What would you like to be added?

Add open telemetry tracing feature for compaction

### Why is this needed?

It could be more convenient for developers to diagnosis performance issues for compactions

### Anything else?

Related to #27606 #27607

### PR DESCRIPTION
**Pull request by congqixia**
Resolves #30167

This PR add tracing for all compaction from the task start in datacoord and execution procedures in datanode.

### CODE REVIEW

**Review by XuanYang-cn — COMMENTED** [review]
Please add tracing for compaction.go `enqueuePlan`

File: internal/datanode/compactor.go [hunk 1]
```diff
@@ -427,15 +430,18 @@ func (t *compactionTask) merge(
 }
 
 func (t *compactionTask) compact() (*datapb.CompactionPlanResult, error) {
+	// span := trace.
```
Comment (XuanYang-cn) [inline internal/datanode/compactor.go:433]: Please remove the comment code

Comment (congqixia) [reply internal/datanode/compactor.go]: done

**Pull request merged**